### PR TITLE
Feature/filtered authentication sources

### PR DESCRIPTION
**Pull request by dwlfrth**
# Description
Reworked and "simplified" the logic of filtering authentication source for a realm

# Impacts
Authentication flow

# Delete branch after merge
YES

### CODE REVIEW

**Comment by dwlfrth** [comment]
Review me

File: lib/pf/config/util.pm [hunk 1]
```diff
@@ -397,60 +397,64 @@ sub portal_hosts {
...
+            $realm_authentication_source = pf::authentication::getAuthenticationSource($source);
         }
     }
-    elsif(exists $ConfigRealm{default} && $realm ne "null"){
-        if(my $source = $ConfigRealm{default}{source}){
-            get_logger->info("Found auth source $source for realm $realm through the default configuration.");
-            $realm_source = pf::authentication::getAuthenticationSource($source);
+    elsif ( exists $ConfigRealm{default} && $realm ne "null" ) {
+        if ( my $source = $ConfigRealm{default}{source} ) {
+            get_logger->info("Found authentication source '$source' for realm '$realm' through the default configuration");
```
Comment (julsemaan) [inline lib/pf/config/util.pm]: through the default realm ?

File: lib/pf/config/util.pm [hunk 2]
```diff
@@ -397,60 +397,64 @@ sub portal_hosts {
...
-    my ($sources, $username, $realm) = @_;
+sub filter_authentication_sources {
+    my ( $sources, $username, $realm ) = @_;
+
+    return @$sources unless ( defined($username) || defined($realm) );
+
+    my $realm_authentication_source = get_realm_authentication_source($username, $realm);
 
-    my $realm_source = get_realm_source($username, $realm);
+    return @$sources unless ( defined($realm_authentication_source->{id}) && $realm_authentication_source->{id} ne "" );
```
Comment (julsemaan) [inline lib/pf/config/util.pm]: Would make sense to check if $realm_authentication_source is defined first IMO

**Comment by dwlfrth** [comment]
Fixaide

**Comment by dwlfrth** [comment]
Sup !

**Pull request merged**